open_spiel/games/dark_chess.cc

```diff
@@ -80,13 +81,57 @@ std::array<bool, chess::kMaxBoardSize * chess::kMaxBoardSize> ComputeObservabili
       auto &piece = board.at(sq);
       if (piece.color == color) {
         size_t index = chess::SquareToIndex(sq, board_size);
-        observability_table[index] = true;
+        if (!publicInfoTable[index]) observability_table[index] = true;
       }
     }
   }
   return observability_table;
 }
 
+// Computes which squares are common knowledge. It does not recognize all of them. Only squares of two opponent
+// pieces of the same type attacking each other
+std::array<bool, chess::kMaxBoardSize * chess::kMaxBoardSize> ComputePublicInfoTable(const chess::ChessBoard& board) {
+  int board_size = board.BoardSize();
+  std::array<bool, chess::kMaxBoardSize * chess::kMaxBoardSize> observability_table{false};
+  board.GenerateLegalMoves([&](const chess::Move &move) -> bool {
+
+    auto from_piece = board.at(move.from);
+    auto to_piece = board.at(move.to);
+
+    if (from_piece.type != chess::PieceType::kEmpty && from_piece.type == to_piece.type) {
```

Comment:
I think I understand the restriction to the same piece type (will the opponent be able to attack me as well?), but then this will make some situations "blind": queen and rook in mutual attack for example.

I think you can do it as computing a boolean mask of piece positions and possible moves per player, and the public info table will be then bool operation && on the tables.